pkg/argocd/update_test.go

```diff
@@ -1079,7 +1079,8 @@ kustomize:
 `
 		app := v1alpha1.Application{
 			ObjectMeta: v1.ObjectMeta{
-				Name: "testapp",
+				Name:      "testapp",
+				Namespace: "argocd",
```

Comment:
Would the test still pass without adding Namespace field, since the behavior should remain the same when the application is in the argocd control plane's namespace?  

If so, we can probably leave some test cases unchanged, some adding "argocd" namespace, and some adding "other" ns, just to cover more cases.

Reply:
OK I'm working on this